tests/components/sentry/test_config_flow.py

```diff
@@ -34,23 +46,94 @@ async def test_form(hass):
         "dsn": "http://public@sentry.local/1",
     }
     await hass.async_block_till_done()
+
     assert len(mock_setup.mock_calls) == 1
     assert len(mock_setup_entry.mock_calls) == 1
 
 
-async def test_form_bad_dsn(hass):
+async def test_integration_already_exists(hass):
+    """Test we only allow a single config flow."""
+    MockConfigEntry(domain=DOMAIN).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_user_flow_bad_dsn(hass):
     """Test we handle bad dsn error."""
     result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
+        DOMAIN, context={"source": SOURCE_USER}
     )
 
     with patch(
-        "homeassistant.components.sentry.config_flow.validate_input",
-        side_effect=BadDsn,
+        "homeassistant.components.sentry.config_flow.Dsn", side_effect=BadDsn,
     ):
         result2 = await hass.config_entries.flow.async_configure(
             result["flow_id"], {"dsn": "foo"},
         )
 
-    assert result2["type"] == "form"
+    assert result2["type"] == RESULT_TYPE_FORM
     assert result2["errors"] == {"base": "bad_dsn"}
+
+
+async def test_user_flow_unkown_exception(hass):
+    """Test we handle any unknown exception error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.sentry.config_flow.Dsn", side_effect=Exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"dsn": "foo"},
+        )
+
+    assert result2["type"] == RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": "unknown"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)